app/src/main/java/com/example/myapplication/MainActivity.kt

```diff
@@ -1,19 +1,62 @@
 package com.example.myapplication
 
+import android.Manifest
+import android.content.pm.PackageManager
 import android.os.Bundle
+import android.util.Log
 import androidx.camera.view.PreviewView
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
 import com.example.myapplication.stuff.CameraActivity
 
+const val MY_PERMISSION_CODE = 100
+
 class MainActivity : CameraActivity() {
 
     override lateinit var previewView: PreviewView
 
     override fun onCreate(savedInstanceState: Bundle?) {
+
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
         previewView = findViewById(R.id.preview_view)
+        if (havePermissionToCamera()
+        ) {
+            ActivityCompat.requestPermissions(
```

Comment:
try to use this approach https://developer.android.com/training/permissions/requesting#request-permission